pkg/apis/crd/v1alpha2/types.go

```diff
@@ -280,3 +270,63 @@ type ExternalIPPoolList struct {
 
 	Items []ExternalIPPool `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPPool defines one or multiple IP sets that can be used for flexible IPAM feature. For instance, the IPs can be
+// allocated to pods according to IP pool specified in deployment annotation.
+type IPPool struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the ExternalIPPool.
+	Spec IPPoolSpec `json:"spec"`
+
+	// Most recently observed status of the pool.
+	Status IPPoolStatus `json:"status"`
+}
+
+type IPPoolSpec struct {
+	// IP Version for this IP pool - either 4 or 6
+	IPVersion int `json:"ipVersion"`
+	// The IP ranges of this IP pool, e.g. 10.10.0.0/24, 10.10.10.2-10.10.10.20, 4001::12-4001::30.
+	IPRanges []IPRange `json:"ipRanges"`
+	// Vlan ID for this IP pool. Default is 0. String-typed for sake of potential autoselect option.
+	Vlan string `json:"vlan,omitempty"`
+}
+
+type IPPoolStatus struct {
+	Usage []IPPoolUsage `json:"usage,omitempty"`
+}
+
+type IPPoolUsage struct {
+	// IP Address this entry is tracking
```

Comment:
if the range is large, will this contain the status of all IPs individually? could that not explode?

Reply:
The idea is to track only allocated or pre-allocated IPs, not the whole range.

Reply:
i was thinking it could be total number of ips used versus free IPs .. something that is being discussed for EgressIPPool https://github.com/antrea-io/antrea/issues/2454#issuecomment-885334259 by @tnqn 

More details about the usage can be exposed via a separate control plane API like how we expose the computed groupmembers of a ClusterGroup.

Reply:
Thanks for the pointer @abhiraut, it would be nice to keep the pool APIs consistent

Reply:
Hi @abhiraut, following the discussion on https://github.com/antrea-io/antrea/pull/2490 - do you still think total usage statistics are needed in this CRD? I agree this provides clear usage picture however it will imply updating the status on spec update, which otherwise is not necessary. Since we plan to enhance antctl with managing these pools, perhaps these stats can be added there.

Reply:
added // TODO for this

Reply:
adding later should be fine IMO